File: android/src/main/java/io/agora/agorartcengine/AgoraRendererView.java
package io.agora.agorartcengine;

import android.view.SurfaceView;
import android.view.View;

import io.flutter.plugin.platform.PlatformView;

/**
 * 封装SurfaceView 给Flutter
 *
 * @author zhengxiaobin
 * @date 2020/6/5
 */
public class AgoraRendererView implements PlatformView {
    private final SurfaceView mSurfaceView;
    private final long uid;

    AgoraRendererView(SurfaceView surfaceView, int uid) {
        this.mSurfaceView = surfaceView;
        this.uid = uid;
    }

    @Override
    public View getView() {
        return mSurfaceView;
    }

    @Override
    public void dispose() {
    }
}

File: android/src/main/java/io/agora/agorartcengine/AgoraRendererView.kt
package io.agora.agorartcengine

import android.view.SurfaceView
import android.view.View
import io.flutter.plugin.platform.PlatformView

/**
 * 封装SurfaceView 给Flutter
 *
 * @author zhengxiaobin
 * @date 2020/6/5
 */
class AgoraRendererView internal constructor(
    private val mSurfaceView: SurfaceView,
    private val viewId: Int
) :
    PlatformView {
    override fun getView(): View {
        return mSurfaceView
    }

    override fun dispose() {}

}